Kernel/Arch/aarch64/ThreadRegisters.h

```diff
@@ -41,6 +44,20 @@ struct ThreadRegisters {
         (void)space;
         TODO_AARCH64();
     }
+
+    void set_spsr_el1(bool is_kernel_process)
+    {
+        Aarch64::SPSR_EL1 saved_program_status_register_el1 = {};
+
+        // Don't mask any interrupts, so all interrupts are enabled when transfering into the new context
+        saved_program_status_register_el1.D = 0;
+        saved_program_status_register_el1.A = 0;
+        saved_program_status_register_el1.I = 0;
+        saved_program_status_register_el1.F = 0;
+
+        saved_program_status_register_el1.M = is_kernel_process ? Aarch64::SPSR_EL1::Mode::EL1h : Aarch64::SPSR_EL1::Mode::EL0t;
```

Comment:
(very nit: i would've put the code move and the behavior change that's in this line here into two commits, so that the larger commit that just moves the code around is behavior-preserving and the commit that changes behavior is small and has a clear commit description saying that, but /shruggie)

Reply:
Yeah, that's definitely more atomic, so I changed that :)